src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,119 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
-
+        resize();
+        if (key == null) {
+            putForNullKey(value);
+        } else {
+            if (table[indexFor(key.hashCode())] == null) {
+                table[indexFor(key.hashCode())] = new Node<K, V>(key, value, null);
+                size++;
+            } else {
+                addNodeInQueue(key, value, indexFor(key.hashCode()));
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int keyIndexInMyTable;
+        keyIndexInMyTable = key == null ? 0 : indexFor(key.hashCode());
+        V valueReturn = null;
+        if (table[keyIndexInMyTable] != null) {
+            Node checked = table[keyIndexInMyTable];
+            while (checked != null) {
+                if (Objects.equals(key, checked.key)) {
+                    return (V) checked.value;
+                }
+                checked = checked.next;
+            }
+        }
+        return valueReturn;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putForNullKey(V value) {
+        if (table[0] == null) {
+            table[0] = new Node(null, value, null);
+            size++;
+        } else {
+            addNodeInQueue(null, value, 0);
+        }
+    }
+
+    private int addNodeInQueue(K key, V value, int index) {
+        Node checked = table[index];
+        while (checked != null) {
+            if (Objects.equals(key, checked.key)) {
+                checked.value = value;
+                return 0;
+            }
+            if (checked.next != null) {
+                checked = checked.next;
+            } else {
+                break;
+            }
+        }
+        checked.next = new Node<K, V>(key, value, null);
+        size++;
+        return 1;
+    }
+
+    private int indexFor(int code) {
```

Comment:
let's make it accept key and if it's null return 0. This way we can get rid of separated methods for key == null